docs/src/generative.md

```diff
@@ -13,15 +13,17 @@ end
 
 ## Interface Description
 
-The *generative* interface consists of two functions:
-- [`gen`](@ref) returns samples (e.g. states, observations and rewards) from a generative POMDP model.
+The *generative* interface consists of two components:
+- The [`gen`](@ref) function or the [`@gen`](@ref) macro returns samples (e.g. states, observations and rewards) from a generative POMDP model.
 - [`initialstate`](@ref) returns a sampled initial state.
 The generative interface is typically used when it is easier to return sampled states and observations rather than explicit distributions as in the [Explicit interface](@ref explicit_doc).
 This type of model is often referred to as a "black-box" model.
 
 In some special cases (e.g. reinforcement learning with [RLInterface.jl](https://github.com/JuliaPOMDP/RLInterface.jl)), an initial observation is needed before any actions are taken. In this case, the [`initialobs`](@ref) function will be used.
 
-## The [`gen`](@ref) function
+## The [`gen`](@ref) function and [`@gen`](@ref) macro
+
+In most cases solvers and simulators should use the [`@gen`](@ref) macro, while problem-writers should implement new methods of the [`gen`](@ref) function.
 
```

Comment:
Should there be an example use of `@gen` in here? I know it is redundant with the docstring but it might still be nice.

Reply:
Sure, just added one